src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/entity/Assessment.kt

```diff
@@ -89,16 +97,32 @@ data class Assessment(
     currentResults.add(criteria)
   }
 
-  fun changeStatus(newStatus: AssessmentStatus) {
-    if (status != newStatus) {
-      assessmentEvents.add(
-        StatusChangedEvent(
-          assessment = this,
-          changes = StatusChange(before = status, after = newStatus),
-        ),
-      )
-      status = newStatus
-      lastUpdatedTimestamp = LocalDateTime.now()
+  private fun Any.label() = this::class.simpleName
+
+  fun performTransition(
+    event: AssessmentLifecycleEvent,
+  ): AssessmentState {
+    val currentStatus = this.status
+    val transition = assessmentStateMachine.with { initialState(currentStatus) }.transition(event)
+    return when (transition) {
+      is Invalid -> {
+        error("Fail to transition Assessment: '${this.id}', triggered by '${transition.event.label()}' from '${transition.fromState.label()}'")
```

Comment:
maybe this should return something rather than erroring? use cases may become apparent